### PR TITLE
[GraphQL/Cursor] DynamicField pagination

### PR DESCRIPTION
**Pull request by amnn**
## Description

Port dynamic field pagination to the new cursor/data/pagination framework.  This is the last pillar supporting paginated multi-gets in the existing data layer, so these all get to disappear following this change.

## Test Plan

```
sui-graphql-rpc$ cargo nextest run
sui-graphql-e2e-tests$ cargo nextest run -j 1 --features pg_integration
```

## Stack
- #15749 
- #15760 
- #15761 
- #15764
- #15777
- #15778
- #15779

### CODE REVIEW

File: crates/sui-graphql-rpc/src/context_data/db_data_provider.rs [hunk 1]
```diff
@@ -414,63 +304,6 @@ impl PgManager {
         Ok(stake)
     }
 
-    pub(crate) async fn fetch_dynamic_fields(
```
Comment (wlmyng) [inline crates/sui-graphql-rpc/src/context_data/db_data_provider.rs:417]: spring cleaning

File: crates/sui-graphql-rpc/src/functional_group.rs [hunk 2]
```diff
@@ -80,12 +83,16 @@ fn functional_groups() -> &'static BTreeMap<(&'static str, &'static str), Functi
             (("Object", "coins"), G::Coins),
             (("Object", "defaultSuinsName"), G::NameService),
             (("Object", "dynamicField"), G::DynamicFields),
-            (("Object", "dynamicFieldConnection"), G::DynamicFields),
+            (("Object", "dynamicObjectField"), G::DynamicFields),
```
Comment (wlmyng) [inline crates/sui-graphql-rpc/src/functional_group.rs:86]: ah good catch here

**Review by wlmyng — APPROVED** [review]
looking good

File: crates/sui-graphql-rpc/src/functional_group.rs (hunk 2 above)
Comment (amnn) [reply crates/sui-graphql-rpc/src/functional_group.rs:86]: The testing support for functional groups is not great (my fault) so it didn't complain before -- we can revisit that post MVP launch.

**Pull request merged**